modules/ListView.js

```diff
@@ -367,48 +417,110 @@
+     * @method __createItemView
      * @private
+     * @param model {Backbone Model} the model to create the view from
+     * @param [noUpdateToIdList=false] if true, the internal order of model ids are not updated
+     * @return {Backbone View} the new item view
      */
-    __removeStaleChildViews: function() {
-      var removedChildViews = [];
+    __createItemView: function(model, noUpdateToIdList) {
+      var itemView,
+        ItemViewClass = this.childView;
+      if (!_.isFunction(this.childView.extend)) {
+        ItemViewClass = this.childView(model);
+      }
+      itemView = new ItemViewClass(this.__generateChildArgs(model));
+      this.registerTrackedView(itemView, { shared: false });
+      this.__modelToViewMap[model[this.__modelId]] = itemView.cid;
+      if (!noUpdateToIdList) {
+        this.__updateOrderedModelIdList();
+      }
+      this.trigger('child-view-added', {model: model, view: itemView});
+      return itemView;
+    },
+
+    /**
+     * Gets all item views that have models that are no longer tracked by modelsToRender
+     * @method __getStaleItemViews
+     * @return {Array} An array of information about stale items. Each object has a 'view' and 'modelId' field
+     * @private
+     */
+    __getStaleItemViews: function() {
+      var staleItemViews = [];
       var modelsWithViews = _.clone(this.__modelToViewMap);
       _.each(this.modelsToRender(), function(model) {
-        var childView = this.getChildViewFromModel(model);
-        if (childView) {
+        var itemView = this.getChildViewFromModel(model);
+        if (itemView) {
           delete modelsWithViews[model[this.__modelId]];
         }
       }, this);
       _.each(modelsWithViews, function(viewId, modelId) {
-        var childView = this.getChildView(viewId);
-        if (childView) {
-          removedChildViews.push(childView);
-          _removeChildView.call(this, childView, modelId);
+        var itemView = this.getChildView(viewId);
+        if (itemView) {
+          staleItemViews.push({ view: itemView, modelId: modelId });
         } else {
           delete this.__modelToViewMap[modelId];
         }
       }, this);
-      return removedChildViews;
+      return staleItemViews;
+    },
+
+    /**
+     * Removes the item views that no longer have models returned by modelsToRender()
+     * @method __removeStaleItemViews
+     * @param [staleItemViewInfo] {Array of objects:
+     *   [{
+     *     view: stale item view,
+     *     modelId: id of model item
+     *   }]} If provided, stale items will not be found, but this array will be used instead.
+     * @private
+     */
+    __removeStaleItemViews: function(staleItemViewInfo) {
+      var view = this;
+      staleItemViewInfo = staleItemViewInfo || this.__getStaleItemViews();
+      _.each(staleItemViewInfo, function(staleViewInfo) {
+        _removeItemView.call(view, staleViewInfo.view, staleViewInfo.modelId);
```

Comment:
Ah, i see the confusion. Yes, it does remove the entry in _modelToViewMap. It does so in _removeItemView.

The delete that happens in __getStaleItemViews removes ids in the map that don't have an assoicated view. This should never happen as that map should be managed carefully. Rather than throw an exception, i deleted the entry there